GOVERNANCE.md

```diff
@@ -0,0 +1,147 @@
+# CoreDNS Governance
+
+## Principles
+
+The CoreDNS community adheres to the following principles:
+
+- Open: CoreDNS is open source, advertised on [our website](https://coredns.io/community).
+- Welcoming and respectful: See [Code of Conduct](CODE-OF-CONDUCT.md).
+- Transparent and accessible: Work and collaboration are done in public.
+- Merit: Ideas and contributions are accepted according to their technical merit and alignment with
+  project objectives, scope, and design principles.
+
+## Project Lead
+
+The CoreDNS project has a project lead.
+
+A project lead in CoreDNS is
+a single person that has a final say in any decision concerning the CoreDNS project.
+
+The term of the project lead is one year, with no term limit restriction.
+
+The project lead is elected by CoreDNS maintainers
+according to an individual's technical merit to CoreDNS project.
+
+The current project lead is identified in the top level [OWNERS](OWNERS) file with the string
+`project lead` and the term behind the name.
+
+
+## Expectations from Maintainers
+
+Every one carries water...
+
+Making a community work requires input/effort from everyone. Maintainers should actively
+participate in Pull Request reviews. Maintainers are expected to respond to assigned Pull Requests
+in a *reasonable* time frame, either providing insights, or assign the Pull Requests to other
+maintainers.
+
+Every Maintainer is listed in the top-level [OWNERS](https://github.com/coredns/coredns/blob/master/OWNERS)
+file, with their Github handle and a possibly obfuscated email address. Everyone in the
+`approvers` list is a Maintainer.
+
+A Maintainer is also listed in a plugin specific OWNERS file.
+
+A Maintainer should be a member of `maintainers@coredns.io`, although this is not a hard requirement.
+
+## Becoming a Maintainer
+
+On successful merge of a significant pull request any current maintainer can reach
+to the author behind the pull request and ask them if they are willing to become a CoreDNS
+maintainer. The email of the new maintainer invitation should be cc'ed to `maintainers@coredns.io`
+as part of the process.
+
+## Changes in Maintainership
+
+If a Maintainer feels she/he can not fulfill the "Expectations from Maintainers", they are free to
+step down.
+
+The CoreDNS organization will never forcefully remove a current Maintainer, unless a maintainer
```

Comment:
How does the decision process of removal work? (e.g. Leader decides, maintainers vote).

Reply:
Ah, never mind.  I see below in the decision making process section, it states  "... and maintainer membership ..."